include/seqan3/std/span

```diff
@@ -185,6 +185,16 @@ public:
     constexpr index_type size_bytes() const noexcept { return span_extent * sizeof(element_type); }
     constexpr bool empty()            const noexcept { return span_extent == 0; }
 
+    constexpr reference front() const noexcept
+    {
+        return this->operator[](0);
```

Comment:
This is real picky, so feel free to ignore it, but on cppreference it is documented as returning `*this->begin()`. Won't make a difference ever right? (I checked because I did not know that span only has a const version of `front()` but of course that makes sense :woman_facepalming:)

Funnily, `back()` is documented as returning `*(this->end()-1)`.. I thought one always avoids this.

Reply:
https://eel.is/c++draft/span.elem says:

```c++
constexpr reference operator[](size_type idx) const;
// Preconditions: idx < size() is true.
// Effects: Equivalent to: return *(data() + idx);

constexpr reference front() const;
// Preconditions: empty() is false.
// Effects: Equivalent to: return *data();

constexpr reference back() const;
// Preconditions: empty() is false.
// Effects: Equivalent to: return *(data() + (size() - 1));

constexpr pointer data() const noexcept;
// Effects: Equivalent to: return data_­;
```